File: src/cli/command.go
package cli

import (
	"fmt"
	"strings"
)

// Command wraps command utility
type Command struct {
	commandName string
}

// NewCommand creates a new Command command builder
func NewCommand() *Command {
	return &Command{}
}

// Command sets command name to check
func (c *Command) SetCommand(name string) *Command {
	c.commandName = name
	return c
}

// Exists generates command to check if a command exists
func (c *Command) Exists() string {
	if c.commandName == "" {
		panic("Command name must be set")
	}
	return fmt.Sprintf("command -v %s", quote(c.commandName))
}

func quote(s string) string {
	if strings.Contains(s, " ") || strings.Contains(s, "$") {
		return fmt.Sprintf("'%s'", strings.ReplaceAll(s, "'", "'\"'\"'"))
	}
	return s
}

File: src/CLI/Command.cs
namespace Enva.CLI;

public class Command
{
    private string? commandName;

    public static Command NewCommand()
    {
        return new Command();
    }

    public Command SetCommand(string name)
    {
        commandName = name;
        return this;
    }

    public string Exists()
    {
        if (string.IsNullOrEmpty(commandName))
        {
            throw new Exception("Command name must be set");
        }
        return $"command -v {Quote(commandName)}";
    }

    private string Quote(string s)
    {
        if (s.Contains(" ") || s.Contains("$"))
        {
            return $"'{s.Replace("'", "'\"'\"'")}'";
        }
        return s;
    }
}
